src/layout/Grid.js

```diff
@@ -137,16 +126,238 @@
+        for(var i = 0;i<this._colCount;i++){
+            xCoords.push((i * this._colSize) + this._offsetX);
+        }
+        var yCoords = [];
+        this._offsetY = this._element.node().getBoundingClientRect().top + (this.gutter()/2);
+        for(var j = 0;j<this._rowCount;j++){
+            yCoords.push((j * this._rowSize) + this._offsetY);
+        }
+        this._gridCoordinates = {x:xCoords,y:yCoords};
+    };
+    
+    Grid.prototype.findCurrentLocation = function (e) {
+        var x_loc,y_loc;
+        for(var x1 = 0;x1<this._gridCoordinates.x.length;x1++){
+            if(parseInt(this._gridCoordinates.x[x1]) <= e.clientX){
+                x_loc = x1;
+            }
+        }
+        for(var y1 = 0;y1<this._gridCoordinates.y.length;y1++){
+            if(parseInt(this._gridCoordinates.y[y1]) <= e.clientY){
+                y_loc = y1;
+            }
+        }
+        if(typeof (x_loc) !== 'undefined' && typeof (y_loc) !== 'undefined'){
+            this._currLoc = [x_loc,y_loc];
+        }
+    };
+    
+    Grid.prototype.overHandle = function (e) {
+        var handle;
+        var handleSize = this._dragCell.handleSize();
+        
+        //Determines which edge cell (if any) this._currLoc is hovering over
+        //An "edge" meaning a dropCell on the exterrior edge of a surface that covers many cells
+        var onSouthEdge = this._dragCell.gridRowSpan() === this._currLoc[1] - this._dragCell.gridRow() + 1;
+        var onNorthEdge = this._dragCell.gridRow() === this._currLoc[1];
+        var onEastEdge = this._dragCell.gridColSpan() === this._currLoc[0] - this._dragCell.gridCol() + 1;
+        var onWestEdge = this._dragCell.gridCol() === this._currLoc[0];
+        
+        var top = this._offsetY + ((this._currLoc[1]) * this._rowSize);
+        var left = this._offsetX + ((this._currLoc[0]) * this._colSize);
+        var width = this._colSize - this.gutter();
+        var height = this._rowSize - this.gutter();
+        
+        var handlesInRange = [];
+        if(top + height >= e.clientY && top + height - handleSize <= e.clientY && onSouthEdge){
+            handlesInRange.push('s');//within SOUTH handle range
+        }
+        else if(top <= e.clientY && top + handleSize >= e.clientY && onNorthEdge){
+            handlesInRange.push('n');//within NORTH handle range
+        }
+        if(left + width >= e.clientX && left + width - handleSize <= e.clientX && onEastEdge){
+            handlesInRange.push('e');//within EAST handle range
+        }
+        else if(left <= e.clientX && left + handleSize >= e.clientX && onWestEdge){
+            handlesInRange.push('w');//within WEST handle range
+        }
+        if(handlesInRange.length){
+            handle = handlesInRange.join('');
+        }
+        return handle;
+    };
+    
+    
+    
+    var hideDragHandles = function(){
+        var handles = document.getElementsByClassName('dragHandle');
```

Comment:
This will hide handles in other grids in design mode on the same page.
Should be changed to be more d3:
```
this._element.selectAll(".dragHandle")
    .style("visibility", "hidden")
;
```